NetCord/Gateway/GatewayClient.cs

```diff
@@ -17,71 +20,385 @@ public partial class GatewayClient : WebSocketClient, IEntity
+    /// </summary>
     public event Func<AutoModerationRule, ValueTask>? AutoModerationRuleUpdate;
+
+    /// <summary>
+    /// Sent when a rule is deleted.
+    /// The inner payload is an <see cref="AutoModerationRule"/> object.
+    /// </summary>
     public event Func<AutoModerationRule, ValueTask>? AutoModerationRuleDelete;
+
+    /// <summary>
+    /// Sent when a rule is triggered and an action is executed (e.g. when a message is blocked).
+    /// </summary>
     public event Func<AutoModerationActionExecutionEventArgs, ValueTask>? AutoModerationActionExecution;
+
+    /// <summary>
+    /// Sent when a new guild channel is created, relevant to the bot.
+    /// The inner payload is an <see cref="IGuildChannel"/> object.
+    /// </summary>
     public event Func<IGuildChannel, ValueTask>? GuildChannelCreate;
+
+    /// <summary>
+    /// Sent when a channel is updated. This is not sent with new messages, those are tracked by <see cref="MessageCreate"/> and <see cref="GuildThreadCreate"/>. This event may reference roles or guild members that no longer exist in the guild.
+    /// The inner payload is an <see cref="IGuildChannel"/> object.
+    /// </summary>
     public event Func<IGuildChannel, ValueTask>? GuildChannelUpdate;
+
+    /// <summary>
+    /// Sent when a channel relevant to the bot is deleted.
+    /// The inner payload is an <see cref="IGuildChannel"/> object.
+    /// </summary>
     public event Func<IGuildChannel, ValueTask>? GuildChannelDelete;
-    public event Func<ChannelPinsUpdateEventArgs, ValueTask>? ChannelPinsUpdate;
+
+    /// <summary>
+    /// Sent when a thread is created, relevant to the bot, or when the current user is added to a thread.
+    /// The inner payload is an <see cref="IGuildChannel"/> object.
+    /// </summary>
     public event Func<GuildThreadCreateEventArgs, ValueTask>? GuildThreadCreate;
+
+    /// <summary>
+    /// Sent when a thread is updated. This is not sent with new messages, those are tracked by <see cref="MessageCreate"/>.
+    /// The inner payload is an <see cref="IGuildChannel"/> object.
+    /// </summary>
     public event Func<GuildThread, ValueTask>? GuildThreadUpdate;
+
+    /// <summary>
+    /// Sent when a thread relevant to the bot is deleted.
+    /// The inner payload is a subset of an <see cref="IGuildChannel"/> object.
+    /// </summary>
     public event Func<GuildThreadDeleteEventArgs, ValueTask>? GuildThreadDelete;
+
+    /// <summary>
+    /// Sent when the current user gains access to a channel.
+    /// </summary>
     public event Func<GuildThreadListSyncEventArgs, ValueTask>? GuildThreadListSync;
+
+    /// <summary>
+    /// Sent when the <see cref="Rest.GuildThreadUser"/> object for the bot is updated. This event is largely just a signal that you are a member of the thread.
+    /// The inner payload is a <see cref="Rest.GuildThreadUser"/> object with a set <see cref="GuildThreadUsersUpdateEventArgs.GuildId"/>.
```

Comment:
The documented payloads are just based off of what discord sends (or documents), rather than what NetCord handles it as. This is actually somewhat clearer since it also implies the contents of the EventArgs object.